### PR TITLE
add mysql backend

### PR DESCRIPTION
**Pull request by WorldWay**

### CODE REVIEW

**Comment by WorldWay** [comment]
add mysql backend feature

1.not yet support the ssl way connecting to mysql
2.has already pass the unit test

**Review by wa5i — CHANGES_REQUESTED** [review]
There is a conflict in Cargo.toml that needs to be resolved.

File: src/storage/mysql/mod.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    // let tls_ca_file = conf.get("tls_ca_file").and_then(|v| v.as_str()).unwrap_or("");
+    // let plaintext_credentials_transmission = conf.get("plaintext_credentials_transmission").and_then(|v| v.as_str()).unwrap_or("");
+    // let max_parralel = conf.get("max_parralel").and_then(|v| v.as_i64()).unwrap_or(128) as i32;
+    // let max_idle_connections = conf.get("max_idle_connections").and_then(|v| v.as_i64()).unwrap_or(0) as i32;
+    // let max_connection_lifetime = conf.get("max_connection_lifetime").and_then(|v| v.as_i64()).unwrap_or(0) as i32;
+    //
+    // now this can not support ssl connection yet. Still need to improve it.
+    let database_url = format!("mysql://{}:{}@{}/{}", username, password, address, database);
+
+    println!("database_url: {}", database_url);
```
[the remark this answers is not in the document]
Comment (WorldWay) [reply src/storage/mysql/mod.rs:42]: got it

File: src/storage/mysql/mod.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    println!();
+    let manager = ConnectionManager::<MysqlConnection>::new(database_url);
+    match r2d2::Pool::builder().build(manager) {
+        Ok(pool) => Ok(pool),
+        Err(e) => {
+            println!("Error: {:?}", e);
```
[the remark this answers is not in the document]
Comment (WorldWay) [reply src/storage/mysql/mod.rs:48]: got it

File: src/storage/mysql/mysql_backend.rs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+#[derive(Queryable, PartialEq, Debug)]
+#[diesel(table_name = vault)]
+pub struct MysqlBackendEntry {
+    pub vault_key: String,
+    pub vault_value: Vec<u8>,
+}
+
+#[derive(Insertable, Deserialize)]
+#[diesel(table_name = vault)]
+pub struct NewMysqlBackendEntry {
```
[the remark this answers is not in the document]
Comment (WorldWay) [reply src/storage/mysql/mysql_backend.rs:35]: It is the common usage of the orm-framework diesel to use two different struct for inserting and selection, in this scence, these two structs are same, I'll have a try to use the same one to solve it

**Comment by WorldWay** [comment]
@wa5i  I have fix these problems, let the workflow run please

**Pull request merged**